Fix for different RSA Encryption in .NET 4.7

lafe opened the pull request:
Starting with .NET 4.7 the `GetAsymmetricAlgorithm` function of a `X509AsymmetricSecurityKey` object returns an `AsymmetricAlgorithm` of type `RSACng` instead of the previous `RSACryptoServiceProvider`. Unfortunately, the `RSACryptoServiceProvider` was hardcoded and therefore, the library breaks on .NET 4.7. This Pull Requests adds compatibility for .NET 4.7 by adding a new `SignatureProvider` for `RSACng` and a Factory that creates the correct `SignatureProvider` based on the encountered `AsymmetricAlgorithm`.

The .NET version of the project has been raised to .NET 4.6 in order to use the `RSACng` object, which has been introduced with .NET 4.6. However, the change is behind a conditional compilation flag "NET46", so it is still possible to build the library for .NET 4.5.

lafe committed:
Updated Readme

lafe committed:
Added support for RSACng

lafe committed:
Added Test to project

lafe committed:
Fixed missing file rename operation in csproj

SharePointRadi commented:
Hello Lafe,

This is a great submission and we thank you for it. We looked at this with the team and need to make an assessment if it affects anything in the PnP-Core library, especially if we need to plan some work on our nugget packaging. We will look at this further as soon as we get to it. 

Thanks!

lafe commented:
@SharePointRadi Any news regarding this PR?

lafe commented:
@SharePointRadi @VesaJuvonen Do you have any update regarding this PR?

lafe committed:
Merge branch 'master' into RsaCngFix

lafe committed:
Fixed typos

lafe committed:
Fixed issue with disposed internal key
- RSACngProxy can be extended in the future to hold a copy of the key, if necessary

rjaakke commented:
Facing the same issue, could really use this PR.
Any news on this?
